packages/decorators/src/decorators/http/request-mapping.ts

```diff
@@ -0,0 +1,205 @@
+ *        {@link Request @Request}
+ *        {@link PathParameter @PathParameter}
+ *        {@link QueryParameter @QueryParameter}
+ *        {@link RequestBody @RequestBody}
+ *
+ * @param path this function should react to. Default: '/'. Can include path (optional) parameters '/user/{userId?}'
+ * @param options see {@link RequestMappingOptions}
+ */
+export function RequestMapping(path?: string, options?: RequestMappingOptions): MethodDecorator {
+    const mergedOptions = {
+        ...defaultOptions,
+        ...options,
+    };
+
+    const validPath = toValidPath(path);
+    const parameters = parsePathWithParameters(validPath);
+
+    return (target: object, propertyName: string | symbol, descriptor: TypedPropertyDescriptor<any>) => {
+        const controller = target.constructor.prototype;
+
+        const method = descriptor?.value;
+        if (!isFunction(method)) {
+            throw new Error('@RequestMapping can only be applied to functions');
+        }
+
+        descriptor.value = async function (...args: unknown[]) {
+            if (!args || args.length === 0) {
+                throw new Error(
+                    `@RequestMapping annotated method ${propertyName.toString()} was provided no arguments`
+                );
+            }
+            const context = args[0];
+            if (!isContext(context)) {
+                throw new Error(
+                    `@RequestMapping annotated method ${propertyName.toString()} was not provided a Context as first argument`
+                );
+            }
+
+            const req = context.req;
+            if (!isHttpRequest(req)) {
+                throw new Error(
+                    `@RequestMapping annotated method ${propertyName.toString()} was provided a context without or invalid http request`
+                );
+            }
+
+            // inject extra path parameters
+            injectParameters(context, parameters);
+
+            handleContextParameter(target, propertyName, context, args);
+            handleRequestParameter(target, propertyName, req, args);
+            handleRequestBodyParameter(target, propertyName, req, args);
+            handleQueryParameters(target, propertyName, req, args);
+            handlePathParameter(target, propertyName, req, args);
+
+            try {
+                const result = await method.apply(this, args);
+                return mergedOptions.ResultMapper(result);
+            } catch (e) {
+                return handleError(e);
+            }
+        };
+
+        // register the mapping in the controller
+        const mappings = (controller.requestMappings as TestableRequestMapping[]) || [];
+
+        const regex = pathWithParametersToRegex(validPath);
+        const mapping = {
+            methods: mergedOptions.methods,
+            regex: regex,
+            func: descriptor.value,
+        };
+        mappings.push(mapping);
+
+        Object.defineProperty(controller, 'requestMappings', {
```

Comment:
Can we use a Symbol instead of `'requestMappings'`?

Reply:
Thanks, has been extracted to methods and refactored.